### PR TITLE
Upgrade to entities 0.14.0-preview.18

### PR DESCRIPTION
**Pull request by zeroZshadow**
#### Description
Upgraded the entities package to latest (0.14.0-preview.18)

#### Tests
* [x] Local deployment
* [x] Cloud deployment

#### Documentation
* [x] Changelog
* [x] Upgrade guide

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@
 ### Breaking Changes
 
 - The minimum supported version of Unity is now 2020.1. [#1459](https://github.com/spatialos/gdk-for-unity/pull/1459)
+- Upgraded Unity Entities to 0.14.0-preview.18. [#1463](https://github.com/spatialos/gdk-for-unity/pull/1463)
+  - Projects now require the `Unity Web Request` build-in package to compile for iOS and Android.
```
Comment (jamiebrynes7) [inline CHANGELOG.md:9]:
```suggestion
  - Projects now require the `Unity Web Request` built-in package to compile for iOS and Android.
```

File: UPGRADE_GUIDE.md [hunk 2]
```diff
@@ -6,6 +6,11 @@
 
 In order to use the GDK for Unity `0.4.0`, you will need to upgrade your project to Unity 2020.1. We test against 2020.1.2f1, but any `2020.1.x` version _should_ work.
 
+### Unity Entities 0.14
+
+Due to our internal upgrade to Unity Entities 0.14.0-preview.18, your projects are now required to include the `Unity Web Request` build-in package.
```
Comment (jamiebrynes7) [inline UPGRADE_GUIDE.md:11]:
```suggestion
Due to our internal upgrade to Unity Entities 0.14.0-preview.18, your projects are now required to include the `Unity Web Request` built-in package.
```

File: workers/unity/Packages/io.improbable.gdk.debug/WorkerInspector/EntityListData.cs [hunk 3]
```diff
@@ -65,10 +69,10 @@ public void RefreshData()
             using (refreshDataMarker.Auto())
             {
                 fullData.Clear();
-                var spatialOSComponentType = world.EntityManager.GetArchetypeChunkComponentType<SpatialEntityId>(true);
+                var spatialOSComponentType = world.EntityManager.GetComponentTypeHandle<SpatialEntityId>(true);
                 var metadataComponentType =
-                    world.EntityManager.GetArchetypeChunkComponentType<Metadata.Component>(true);
-                var ecsEntityType = world.EntityManager.GetArchetypeChunkEntityType();
+                    world.EntityManager.GetComponentTypeHandle<Metadata.Component>(true);
+                var ecsEntityType = world.EntityManager.GetEntityTypeHandle();
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.debug/WorkerInspector/EntityListData.cs:75]: Oh thank god for these renames.

**Review by jamiebrynes7 — APPROVED** [review]

File: UPGRADE_GUIDE.md [hunk 4]
```diff
@@ -6,6 +6,11 @@
 
 In order to use the GDK for Unity `0.4.0`, you will need to upgrade your project to Unity 2020.1. We test against 2020.1.2f1, but any `2020.1.x` version _should_ work.
 
+### Unity Entities 0.14
+
+Due to our internal upgrade to Unity Entities 0.14.0-preview.18, your projects are now required to include the `Unity Web Request` built-in package.
```
Comment (paulbalaji) [inline UPGRADE_GUIDE.md:11]:
```suggestion
Due to our internal upgrade to Unity Entities `0.14.0-preview.18`, your projects are now required to include the `Unity Web Request` built-in package.
```

**Review by paulbalaji — APPROVED** [review]

**Pull request merged**